src/minimega/vnc_cli.go

```diff
@@ -27,72 +28,106 @@ If playback is selected, the specified file (created using vnc record) will be
 read and processed as a sequence of time-stamped mouse/keyboard events to send
 to the specified VM.`,
 		Patterns: []string{
-			"vnc",
-			"vnc <kb,fb> <record,> <host> <vm id or name> <filename>",
-			"vnc <kb,fb> <norecord,> <host> <vm id or name>",
-			"vnc <playback,> <host> <vm id or name> <filename>",
-			"vnc <noplayback,> <host> <vm id or name>",
+			"vnc <kb,fb> <record,> <vm name> <filename>",
+			"vnc <kb,fb> <norecord,> <vm name>",
+			"vnc <playback,> <vm name> <filename>",
+			"vnc <noplayback,> <vm name>",
 		},
-		Call: wrapSimpleCLI(cliVNC),
+		Call: wrapVMTargetCLI(cliVNC),
 	},
-	{ // clear vnc
+	{
 		HelpShort: "reset VNC state",
 		HelpLong: `
 Resets the state for VNC recordings. See "help vnc" for more information.`,
 		Patterns: []string{
 			"clear vnc",
 		},
-		Call: wrapSimpleCLI(func(_ *minicli.Command, _ *minicli.Response) error {
-			return vncClear()
-		}),
+		Call: wrapBroadcastCLI(cliVNCClear),
+	},
+	{
+		HelpShort: "list all running vnc playback/recording instances",
+		HelpLong: `
+List all running vnc playback/recording instances. See "help vnc" for more information.`,
+		Patterns: []string{
+			"vnc",
+		},
+		Call: wrapBroadcastCLI(cliVNCList),
 	},
 }
 
+// List all active recordings and playbacks
+func cliVNCList(c *minicli.Command, resp *minicli.Response) error {
+	resp.Header = []string{"host", "name", "id", "type", "filename"}
+	resp.Tabular = [][]string{}
+
+	for _, v := range vncKBRecording {
+		if inNamespace(v.Vm) {
+			resp.Tabular = append(resp.Tabular, []string{
+				hostname, v.Vm.Name, strconv.Itoa(v.Vm.ID),
+				"record kb",
+				v.file.Name(),
+			})
+		}
+	}
+
+	for _, v := range vncFBRecording {
+		if inNamespace(v.Vm) {
+			resp.Tabular = append(resp.Tabular, []string{
+				hostname, v.Vm.Name, strconv.Itoa(v.Vm.ID),
+				"record fb",
+				v.file.Name(),
+			})
+		}
+	}
+
+	for _, v := range vncKBPlaying {
+		if inNamespace(v.Vm) {
+			resp.Tabular = append(resp.Tabular, []string{
+				hostname, v.Vm.Name, strconv.Itoa(v.Vm.ID),
+				"playback kb",
+				v.file.Name(),
+			})
+		}
+	}
+
+	return nil
+}
+
 func cliVNC(c *minicli.Command, resp *minicli.Response) error {
-	host := c.StringArgs["host"]
-	vm := c.StringArgs["vm"]
 	fname := c.StringArgs["filename"]
 
-	if host == Localhost {
-		host = hostname
+	vm, err := vms.FindKvmVM(c.StringArgs["vm"])
+	if err != nil {
+		return fmt.Errorf("vm %s not found", c.StringArgs["vm"])
 	}
 
 	if c.BoolArgs["record"] && c.BoolArgs["kb"] {
 		// Starting keyboard recording
-		return vncRecordKB(host, vm, fname)
+		return vncRecordKB(vm.Name, fname)
```

Comment:
We already have a reference to the VM -- we should pass it in here and then we don't have to find it again later in NewVNCClient.